src/mlpack/core/hpt/cv_function_impl.hpp

```diff
@@ -65,6 +70,26 @@ template<typename CVType,
          typename MLAlgorithm,
          size_t TotalArgs,
          typename... BoundArgs>
+void CVFunction<CVType, MLAlgorithm, TotalArgs, BoundArgs...>::Gradient(
+    const arma::mat& parameters,
+    arma::mat& gradient)
+{
+  gradient = arma::mat(arma::size(parameters));
+  arma::mat increasedParameters = parameters;
+  for (size_t i = 0; i < parameters.n_rows; ++i)
+  {
+    double delta = std::max(std::abs(parameters(i)) * relativeDelta, minDelta);
+    increasedParameters(i) += delta;
+    gradient(i) = (Evaluate(increasedParameters) - Evaluate(parameters)) /
+        delta;
+    increasedParameters(i) = parameters(i);
```

Comment:
Oh, sure. I'm surprised why I haven't cached `Evaluate(parameters)` it in the first place. But I guess we should keep `increasedParameters` as it is since we accept `parameters` with the `const` qualifier.